pkg/validators/am0004_icon_base64.go

```diff
@@ -0,0 +1,29 @@
+package validators
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+)
+
+func init() {
+	Registry.Add(AM0004)
+}
+
+var AM0004 = utils.Validator{
+	Code:        "AM0004",
+	Name:        "icon_base64",
+	Description: "Ensure that `icon` in Addon metadata is rightfully base64 encoded",
+	Runner:      ValidateIconBase64,
+}
+
+// ValidateIconBase64 validates 'icon' in the addon metadata is rightfully base64 encoded
+func ValidateIconBase64(metabundle utils.MetaBundle) (bool, string, error) {
```

Comment:
It would be nice if we would also validate that the `icon` is a valid png.

As enforced in our schema: https://github.com/mt-sre/managed-tenants-cli/blob/main/managedtenants/data/metadata.schema.yaml#L22

It's unfortunate that kubebuilder doesn't have a base64 + mediatype annotation, but we should enforce this as otherwise, we can break the UI and user experience when installing an addon.